core/templates/dev/head/exploration_editor/RouterServices.js

```diff
@@ -137,34 +158,26 @@ oppia.factory('routerService', [
       },
       navigateToMainTab: function(stateName) {
         _savePendingChanges();
-
         if (_getCurrentStateFromLocationPath() === stateName) {
           return;
         }
 
-        var _actuallyNavigate = function(newStateName) {
-          if (newStateName) {
-            editorContextService.setActiveStateName(newStateName);
-          }
-          $location.path('/gui/' + editorContextService.getActiveStateName());
-          $window.scrollTo(0, 0);
-        };
-
         if (_tabs.active === MAIN_TAB) {
           $('.oppia-editor-cards-container').fadeOut(function() {
-            _actuallyNavigate(stateName);
+            _actuallyNavigate(SLUG_GUI, stateName);
             $rootScope.$apply();
             $timeout(function() {
               $('.oppia-editor-cards-container').fadeIn();
             }, 150);
           });
         } else {
-          _actuallyNavigate(stateName);
+          _actuallyNavigate(SLUG_GUI, stateName);
         }
       },
       navigateToPreviewTab: function() {
-        _savePendingChanges();
-        $location.path('/preview');
+        if (_tabs.active !== PREVIEW_TAB) {
```

Comment:
Oh, interesting. I'm not sure. How about, for now, just restoring it anyway, but adding a comment above it:

  TODO(oparry): Determine whether this is necessary, since there's already a $rootScope.$broadcast('externalSave') in line 56

(note that the name next to the TODO refers to who is reporting it, not who should do it). Also, please file an issue on GitHub about this so we don't forget to look into it.

Thanks!